### ISSUE TITLE
First render only produces style background

### ISSUE THREAD

**Issue by averas** [mapbox/mapbox-gl-native#7719]
**Platform: Linux (Ubuntu)**
**Mapbox SDK version: node-mapbox-gl-native 3.4.2**

I'm using xvfb for headless rendering of raster map tiles using GL styles which works very good,  except for one issue: Every time I start my server the _first_ render call returns an image buffer with only the style background present (a specific color, so it's not blank). All layers on top are missing. The next render pass returns an image buffer in accordance with the style and from that point everything works fine.

I'm not sure whether this is a known issue (couldn't find any related issues), an issue with my setup (for example with xvfb), or whether it's simply user error. My implementation however, is pretty much ripped directly from the examples.

**Comment by 1ec5** [comment]
I also see this occasionally with the macOS SDK – the next frame renders fine – but I’m not sure if it’s related.

**Comment by averas** [comment]
@1ec5 Interesting. Probably nothing obvious then. I'll go ahead and setup a contained test case demonstrating the issue.

**Comment by averas** [comment]
@1ec5 Alright. I have a test case for this now, and while doing it I isolated some suspicious behaviour.

```bash
$ git clone https://github.com/averas/first-render-fail.git
$ cd first-render-fail
$ npm install
$ npm run start
https://b.tiles.mapbox.com/v4/mapbox.mapbox-streets-v7/15/17752/16383.vector.pbf
https://b.tiles.mapbox.com/v4/mapbox.mapbox-streets-v7/15/17752/16384.vector.pbf
https://b.tiles.mapbox.com/v4/mapbox.mapbox-streets-v7/15/17752/9597.vector.pbf
```

This will run two render passes which will produce image0.png and image1.png, the first one only contains the background where the second one also contains the water layer. The output from the script are the vector tiles fetched, and here's where the suspicious behaviour lies. The first two tiles are nowhere near the provided zoom/center, so it seems these settings are off for the first render pass. The second render pass, fetches the correct vector tile and also produces a correct image.

After this analysis I actually believe this is the same issue as #7351 which is fixed by #7413, however, the latest version in NPM is 3.4.2 which does not contain this fix.

**Comment by averas** [comment]
Yes, I confirm that this is in fact a dupe of #7351 and will be fixed in npm as soon as 3.4.4 is published.
